# Week 1 HW Submission

chongjunkim opened the pull request:
# Week 1 HW Submission

Please fill out the information below in order to complete your assignment. Feel free to update this comment later if necessary.

* Comfort rating on this assignment (1-5): 5
* Completion rating on this assignment (complete/incomplete): complete
* Github handles of code I've reviewed: n/a

Commit:
Week 1 Assignment: ES6

Commit:
npm test

Commit:
fix

bhague1281 commented on src/components/DivElement.js:
```diff
@@ -1,6 +1,13 @@
+  constructor(content) {
+    super(content);
+    this.tag = 'div';
+    this.content = content;
```
While this works, I would refrain from explicitly defining `this.tag` and `this.content` in this scenario. Reason being is that we can offload that work to the `HTMLElement` constructor.

There can be scenarios where we start editing the `HTMLElement` constructor, expecting those changes to be applied to child classes. Here's an example: I decide to alter all HTMLElements so that the tag is always set to lowercase:

```js
class HTMLElement {
  constructor(tag, content) {
    this.tag = tag.toLowerCase();
    this.content = content;
  }

  ...
}
```

Instead, call the parent constructor with `super()`, and let it take care of `tag` and `content` for you 😃. Here are some more examples: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super

bhague1281 reviewed (approved):
Great job! I made a comment on your `DivElement` class. Feel free to resubmit that portion if you wish.

Commit:
Update

Used super() as suggested in the DivElement class. I've also updated some code to match your example and also updated the json file to pass the npm test 100%.
